ckan/logic/action/get.py

```diff
@@ -1884,9 +1884,12 @@ def package_search(context, data_dict):
         include_drafts = asbool(data_dict.pop('include_drafts', False))
         data_dict.setdefault('fq', '')
         if not include_private:
-            data_dict['fq'] = '+capacity:public ' + data_dict['fq']
+            data_dict['fq'] = 'capacity:public' if not data_dict['fq'] \
+                else 'capacity:public +' + data_dict['fq']
         if include_drafts:
             data_dict['fq'] += ' +state:(active OR draft)'
+        if not include_drafts and not context.get('sysadmin'):
+            data_dict['fq'] += ' -state:draft'
```

Comment:
We already have default filter by active state  - https://github.com/ckan/ckan/blob/master/ckan/lib/search/query.py#L260 
How is this different from one from the link?

Reply:
I added this line because a test was breaking--I think it was https://github.com/ckan/ckan/blob/cf7f4cc04204a4bb14e140b0e7e9297787f35f6a/ckan/tests/logic/action/test_get.py#L1278, testing that you can't search with `fq = "state:draft"` and see draft datasets. Previously, this test passed, because the `state:draft` part of the query was just ignored as described above. Adding this line fixed it.

Looking at the test again, though, I think something is still wrong. The test doesn't include `'sysadmin': True` as in https://github.com/ckan/ckan/pull/4962/files/b1a31168334a54092938b1dd62a956ba475beac2#diff-c92cdbf4ff7391d90f31341452b90303R1518, and that's why the search in the test doesn't output the draft datasets.